pkg/cmd/pulumi/util.go

```diff
@@ -975,6 +977,26 @@ func addUpdatePlanMetadataToEnvironment(env map[string]string, updatePlan bool)
 	env[backend.UpdatePlan] = strconv.FormatBool(updatePlan)
 }
 
+type EscEnvironmentMetadata struct {
+	Name string `json:"name"`
+}
+
+// addEscMetadataToEnvironment populates the environment metadata bag with the ESC environments
+// used as part of the stack update.
+func addEscMetadataToEnvironment(env map[string]string, escEnvironments []string) {
+	envs := make([]EscEnvironmentMetadata, len(escEnvironments))
+	for i, s := range escEnvironments {
+		envs[i] = EscEnvironmentMetadata{Name: s}
```

Comment:
Why not a `[]string` instead of `[]EscEnvironmentMetadata` since there's only 1 key?

Reply:
I talked with @dschaller about this and we would like to eventually add a `resolvedRevision` to this so we can track which revision was used. this will be a followup - I'll create an issue for this